src/runtime_src/driver/xclng/tools/xbutil/xbutil.cpp

```diff
@@ -1479,3 +1493,174 @@ int xcldev::xclP2p(int argc, char *argv[])
 
     return ret;
 }
+
+/*
+ * m2mtest
+ */
+static void m2m_free_unmap_bo(xclDeviceHandle handle, unsigned boh, 
+auto * &boptr, size_t boSize)
+{
+    if(boh != NULLBO)
+        xclFreeBO(handle, boh);
+    if(boptr != nullptr)
+        munmap(boptr, boSize);
+}
+
+static int m2m_alloc_init_bo(xclDeviceHandle handle, char * &boptr, 
+size_t boSize, int bank, char pattern)
+{
+    unsigned boh = NULLBO;
+    boh = xclAllocBO(handle, boSize, XCL_BO_DEVICE_RAM, bank);
+    if (boh == NULLBO) {
+        std::cout << "Error allocating BO" << std::endl;
+        return -ENOMEM;
+    }
+    boptr = (char*) xclMapBO(handle, boh, true);
+    if (boptr == nullptr) {
+        std::cout << "Error mapping BO" << std::endl;
+        m2m_free_unmap_bo(handle, boh, boptr, boSize);
+        return -EINVAL;
+    }
+    memset(boptr, pattern, boSize);
+    return boh;
+}
+static int m2mtest_bank(xclDeviceHandle handle, uuid_t uuid, int bank_a, int bank_b)
+{
+    unsigned boSrc = NULLBO;
+    unsigned boTgt = NULLBO;
+    char *boSrcPtr = nullptr;
+    char *boTgtPtr = nullptr;
+
+    const size_t boSize = 256L * 1024 * 1024;
+    if (xclOpenContext(handle, uuid, -1, true)) {
+        std::cout << "ERROR: Unable to lockdown xclbin" << std::endl;
+        return -EINVAL;
+    }
+
+    //Allocate and init boSrc
+    boSrc = m2m_alloc_init_bo(handle, boSrcPtr, boSize, bank_a, 'A');
+    if(xclSyncBO(handle, boSrc, XCL_BO_SYNC_BO_TO_DEVICE, boSize, 0)) {
```

Comment:
why not xclSyncBO inside m2m_alloc_init_bo()?

Reply:
I am cleaning up srcbo as well when cleaning up syncBO for tgt. I am not passing any srcbo information in "m2m_alloc_init_bo" so I won't be able to clean that up